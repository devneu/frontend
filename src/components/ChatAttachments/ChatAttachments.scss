.chat-attachments {
  padding: 0 10px;

  .file-list {
    display: flex;
    flex-direction: column;
  }

  .file-item {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-bottom: 10px;
  }

  .file-info {
    padding-left: 5px;
    font-size: 12px;
  }

  .file-icon {

    width: 30px;


  }
}
