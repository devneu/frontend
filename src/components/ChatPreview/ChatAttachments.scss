.attachments {
  padding: 10px 0 0 10px;

  .document-list {
    display: flex;
    flex-direction: column;
  }

  .file-item {
    display: flex;
    flex-direction: row;
  }

  .file-info {
    padding: 12px 0;
  }

  .file-icon {
    padding: 10px;
    width: 55px;
    height: 55px;

  }
}
