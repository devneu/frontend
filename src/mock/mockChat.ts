export const photos = [
  {
    img: 'https://images.unsplash.com/photo-1469854523086-cc02fe5d8800?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8M3x8fGVufDB8fHx8&w=1000&q=80',
  },
  {
    img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSV_EfZmXNnuKrtvDecfz6B7r955J6X7EFTsw&usqp=CAU',
  },
  {
    img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSrOELmQBbTGeOYYBHmT6UxFVma2GOPE-6kpg&usqp=CAU',
  },
  {
    img: 'https://sourceduk.com/wp-content/uploads/2018/10/girlgoneabroad.jpg',
  },
  {
    img: 'https://www.workandtravelgroup.com/wp-content/uploads/wtg-group-2.jpg',
  },
  {
    img: 'https://www.moneycrashers.com/wp-content/uploads/2018/10/travel-world-map-1068x713.jpg',
  },
];

export const attachments = [
  {
    name: 'pdf',
    img: 'https://www.iconpacks.net/icons/2/free-pdf-icon-3385.png',
    description: 'how to create a case study',
  },
  {
    name: 'zip',
    img: 'https://www.iconpacks.net/icons/2/free-zip-icon-1519-thumb.png',
    description: 'Project Documents',
  },
  {
    name: 'word',
    img: 'https://www.pngkey.com/png/full/210-2104640_imagen-microsoft-office-365-word-logo.png ',
    description: 'Competitor Analysis Template',
  },
];

export const avatar = {
  img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQnPGqX4s6HDBoVTLwIhy3fFmdxvMiDIfUtdA&usqp=CAU',
};

export const groupData = {
  img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQnPGqX4s6HDBoVTLwIhy3fFmdxvMiDIfUtdA&usqp=CAU',
  name: 'Nasir Uddin',
  description: 'Ux Designer',
  icon: 'https://i.pinimg.com/originals/44/9d/d1/449dd1f38a8bd87b41f7b960db0fefff.png',
  memberImgs: [
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQnPGqX4s6HDBoVTLwIhy3fFmdxvMiDIfUtdA&usqp=CAU',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSgnq9pVEA16U0vH0nT0UeFY9vrTn99Za2a7QWub_dBpXSYTCZtBQULWaaRJ4ENFreEmPc&usqp=CAU',
    },
    {
      img: 'https://international-home.net/wp-content/uploads/2018/12/matureman1-512.png',
    },
    {
      img: 'https://www.w3schools.com/howto/img_avatar2.png',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSgnq9pVEA16U0vH0nT0UeFY9vrTn99Za2a7QWub_dBpXSYTCZtBQULWaaRJ4ENFreEmPc&usqp=CAU',
    },
    {
      img: 'https://international-home.net/wp-content/uploads/2018/12/matureman1-512.png',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQVxhAxJ4D7MOeTTj6kR9PBeZonW5HM7giKjTbEmR-HMBwf3G1VqGnlwpO1kWrdyIZu8_U&usqp=CAU',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSgnq9pVEA16U0vH0nT0UeFY9vrTn99Za2a7QWub_dBpXSYTCZtBQULWaaRJ4ENFreEmPc&usqp=CAU',
    },
    {
      img: 'https://international-home.net/wp-content/uploads/2018/12/matureman1-512.png',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQVxhAxJ4D7MOeTTj6kR9PBeZonW5HM7giKjTbEmR-HMBwf3G1VqGnlwpO1kWrdyIZu8_U&usqp=CAU',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSgnq9pVEA16U0vH0nT0UeFY9vrTn99Za2a7QWub_dBpXSYTCZtBQULWaaRJ4ENFreEmPc&usqp=CAU',
    },
    {
      img: 'https://www.w3schools.com/howto/img_avatar.png',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQVxhAxJ4D7MOeTTj6kR9PBeZonW5HM7giKjTbEmR-HMBwf3G1VqGnlwpO1kWrdyIZu8_U&usqp=CAU',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSgnq9pVEA16U0vH0nT0UeFY9vrTn99Za2a7QWub_dBpXSYTCZtBQULWaaRJ4ENFreEmPc&usqp=CAU',
    },
    {
      img: 'https://international-home.net/wp-content/uploads/2018/12/matureman1-512.png',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQVxhAxJ4D7MOeTTj6kR9PBeZonW5HM7giKjTbEmR-HMBwf3G1VqGnlwpO1kWrdyIZu8_U&usqp=CAU',
    },
    {
      img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSgnq9pVEA16U0vH0nT0UeFY9vrTn99Za2a7QWub_dBpXSYTCZtBQULWaaRJ4ENFreEmPc&usqp=CAU',
    },
    {
      img: 'https://www.w3schools.com/w3images/avatar6.png',
    },
  ],
};
