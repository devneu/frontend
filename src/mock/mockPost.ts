export const createMockPost = (options = {} as any) => ({
  id: options.id || 999,
  img: 'https://dummyimage.com/700x350/dee2e6/6c757d.jpg',
  title: 'DEV-17.com',
  date: 'September 23, 2010',
  text: `Science is an enterprise that should be cherished as an activity of the free human mind. 
    Because it transforms who we are, how we live, and it gives us an understanding of our place in the universe. 
    The universe is large and old, and the ingredients for life as we know it are everywhere, so there's no reason to think that Earth would be unique in that regard. Whether of not the life became intelligent is a different question, and we'll see if we find that. 
    If you get asteroids about a kilometer in size, those are large enough and carry enough energy into our system to disrupt transportation, communication, the food chains, and that can be a really bad day on Earth.`,
  articleText: `
    # I have odd cosmic thoughts every day
    
    For me, the most fascinating interface is Twitter. I have odd cosmic thoughts every day and I realized I could hold them to myself or share them with people who might be interested.
    
    Venus has a runaway greenhouse effect. I kind of want to know what happened there because we're twirling knobs here on Earth without knowing the consequences of it. Mars once had running water. It's bone dry today. Something bad happened there as well.
    `,
  tag: ['web', 'js'],
});

export const createMockPosts = () => [
  {
    id: 1,
    img: 'https://dummyimage.com/700x350/dee2e6/6c757d.jpg',
    title: 'PostPage Title',
    text: `Science is an enterprise that should be cherished as an activity of the free human mind. 
    Because it transforms who we are, how we live, and it gives us an understanding of our place in the universe. 
    The universe is large and old, and the ingredients for life as we know it are everywhere, so there's no reason to think that Earth would be unique in that regard. Whether of not the life became intelligent is a different question, and we'll see if we find that. 
    If you get asteroids about a kilometer in size, those are large enough and carry enough energy into our system to disrupt transportation, communication, the food chains, and that can be a really bad day on Earth.`,
    date: 'September 23, 2010',
    articleText: `
# I have odd cosmic thoughts every day

For me, the most fascinating interface is Twitter. I have odd cosmic thoughts every day and I realized I could hold them to myself or share them with people who might be interested.

Venus has a runaway greenhouse effect. I kind of want to know what happened there because we're twirling knobs here on Earth without knowing the consequences of it. Mars once had running water. It's bone dry today. Something bad happened there as well.
    `,
    tag: ['Web Design', 'Freebies'],
  },
  {
    id: 2,
    img: 'https://dummyimage.com/700x350/dee2e6/6c757d.jpg',
    title: 'PostPage Title',
    text: `Science is an enterprise that should be cherished as an activity of the free human mind. 
    Because it transforms who we are, how we live, and it gives us an understanding of our place in the universe. 
    The universe is large and old, and the ingredients for life as we know it are everywhere, so there's no reason to think that Earth would be unique in that regard. Whether of not the life became intelligent is a different question, and we'll see if we find that. 
    If you get asteroids about a kilometer in size, those are large enough and carry enough energy into our system to disrupt transportation, communication, the food chains, and that can be a really bad day on Earth.`,
    date: 'September 23, 2010',
    articleText: `
# I have odd cosmic thoughts every day

For me, the most fascinating interface is Twitter. I have odd cosmic thoughts every day and I realized I could hold them to myself or share them with people who might be interested.

Venus has a runaway greenhouse effect. I kind of want to know what happened there because we're twirling knobs here on Earth without knowing the consequences of it. Mars once had running water. It's bone dry today. Something bad happened there as well.
    `,
    tag: ['Web Design', 'Freebies'],
  },
  {
    id: 3,
    img: 'https://dummyimage.com/700x350/dee2e6/6c757d.jpg',
    title: 'PostPage Title',
    text: `Science is an enterprise that should be cherished as an activity of the free human mind. 
    Because it transforms who we are, how we live, and it gives us an understanding of our place in the universe. 
    The universe is large and old, and the ingredients for life as we know it are everywhere, so there's no reason to think that Earth would be unique in that regard. Whether of not the life became intelligent is a different question, and we'll see if we find that. 
    If you get asteroids about a kilometer in size, those are large enough and carry enough energy into our system to disrupt transportation, communication, the food chains, and that can be a really bad day on Earth.`,
    date: 'September 23, 2010',
    articleText: `
# I have odd cosmic thoughts every day

For me, the most fascinating interface is Twitter. I have odd cosmic thoughts every day and I realized I could hold them to myself or share them with people who might be interested.

Venus has a runaway greenhouse effect. I kind of want to know what happened there because we're twirling knobs here on Earth without knowing the consequences of it. Mars once had running water. It's bone dry today. Something bad happened there as well.
    `,
    tag: ['Web Design', 'Freebies'],
  },
  {
    id: 4,
    img: 'https://dummyimage.com/700x350/dee2e6/6c757d.jpg',
    title: 'PostPage Title',
    text: `Science is an enterprise that should be cherished as an activity of the free human mind. 
    Because it transforms who we are, how we live, and it gives us an understanding of our place in the universe. 
    The universe is large and old, and the ingredients for life as we know it are everywhere, so there's no reason to think that Earth would be unique in that regard. Whether of not the life became intelligent is a different question, and we'll see if we find that. 
    If you get asteroids about a kilometer in size, those are large enough and carry enough energy into our system to disrupt transportation, communication, the food chains, and that can be a really bad day on Earth.`,
    date: 'September 23, 2010',
    articleText: `
# I have odd cosmic thoughts every day

For me, the most fascinating interface is Twitter. I have odd cosmic thoughts every day and I realized I could hold them to myself or share them with people who might be interested.

Venus has a runaway greenhouse effect. I kind of want to know what happened there because we're twirling knobs here on Earth without knowing the consequences of it. Mars once had running water. It's bone dry today. Something bad happened there as well.
    `,
    tag: ['Web Design', 'Freebies'],
  },
];
