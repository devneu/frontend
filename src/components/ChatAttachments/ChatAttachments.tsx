import React from 'react';
import Paragraph from 'antd/es/typography/Paragraph';
import { Button } from 'antd';
import './ChatAttachments.scss';
import { Attachments } from '../../types/ChatPreview.interface';

interface ChatAttachmentsProps {
  attachments: Attachments[];
}

const ChatAttachments = ({ attachments }: ChatAttachmentsProps) => (
  <div className="chat-attachments">
    <Paragraph className="attachments-title" strong>
      Attachments
    </Paragraph>
    <div className="file-list">
      <Paragraph className="file-source">Source file</Paragraph>
      {attachments.map((attachment) => (
        <div className="file-item" key={attachment.img}>
          <img className="file-icon" src={attachment.img} alt="#" />
          <div className="file-info">{attachment.description}</div>
        </div>
      ))}
    </div>
    <Button className="show-all" type="link" block>
      View all
    </Button>
  </div>
);

export default ChatAttachments;
