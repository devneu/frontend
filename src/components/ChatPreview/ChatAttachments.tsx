import React from 'react';
import Paragraph from 'antd/es/typography/Paragraph';
import { Button } from 'antd';
import './ChatAttachments.scss';
import { ChatData } from '../../types/ChatPreview.interface';

interface attachmentsProps {
  attachments: ChatData[];
}

const ChatAttachments = ({ attachments }: attachmentsProps) => (
  <div className="chat-attachments">
    <Paragraph>Attachments</Paragraph>
    <div className="file-list">
      {attachments.map((attachment: any) => (
        <div className="file-item">
          <img className="file-icon" src={attachment.img} alt="#" />
          <div className="file-info">{attachment.description}</div>
        </div>
      ))}
    </div>
    <Button className="show-all">View all</Button>
  </div>
);

export default ChatAttachments;
