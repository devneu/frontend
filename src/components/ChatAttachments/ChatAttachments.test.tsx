import React from 'react';
import renderer from 'react-test-renderer';
import ChatAttachments from './ChatAttachments';
import { attachments } from '../../mock/mockChat';

describe('ChatAttachments', () => {
  it('Should successfully render', () => {
    const wrapper = renderer.create(<ChatAttachments attachments={attachments} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
