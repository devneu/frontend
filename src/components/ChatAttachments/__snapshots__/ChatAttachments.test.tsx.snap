// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatAttachments Should successfully render 1`] = `
<div
  className="chat-attachments"
>
  <div
    className="ant-typography attachments-title"
    style={
      Object {
        "WebkitLineClamp": undefined,
      }
    }
  >
    <strong>
      Attachments
    </strong>
  </div>
  <div
    className="file-list"
  >
    <div
      className="ant-typography file-source"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Source file
    </div>
    <div
      className="file-item"
    >
      <img
        alt="#"
        className="file-icon"
        src="https://www.iconpacks.net/icons/2/free-pdf-icon-3385.png"
      />
      <div
        className="file-info"
      >
        how to create a case study
      </div>
    </div>
    <div
      className="file-item"
    >
      <img
        alt="#"
        className="file-icon"
        src="https://www.iconpacks.net/icons/2/free-zip-icon-1519-thumb.png"
      />
      <div
        className="file-info"
      >
        Project Documents
      </div>
    </div>
    <div
      className="file-item"
    >
      <img
        alt="#"
        className="file-icon"
        src="https://www.pngkey.com/png/full/210-2104640_imagen-microsoft-office-365-word-logo.png "
      />
      <div
        className="file-info"
      >
        Competitor Analysis Template
      </div>
    </div>
  </div>
  <button
    className="ant-btn ant-btn-link ant-btn-block show-all"
    onClick={[Function]}
    type="button"
  >
    <span>
      View all
    </span>
  </button>
</div>
`;
